File: tutorials/testing/java/tut_us_map_p2.java
import java.util.ArrayList;
import java.util.Map;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.USState;
import bridges.data_src_dependent.USCounty;
import bridges.base.SLelement;
import bridges.base.USMap;
import bridges.base.Color;


// This tutorial illustrates how to access the data of the US map with state
// boundaries with different colors for states and its boundaries
// It also illustrates how to draw all the counties 

public class tut_us_map_p2 {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
		// set title, description
		bridges.setTitle("Tutorial: Drawing US Map with State Boundaries, and State Counties");
		bridges.setDescription("All US states, All Counties");

		// get data source object
		DataSource ds = bridges.getDataSource();

		// get us map data - states and boundaries
		ArrayList<USState> map_data = ds.getUSMapCountyData ();

		// create a USMap object with the map data
		USMap us_maps = new USMap(map_data);

		// set data structure
		bridges.setDataStructure(us_maps);

		// visualize
		bridges.visualize();
	}
}

File: tutorials/testing/c++/tut_us_map_p2.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/USState.h"
#include "data_src/USCounty.h"
#include "USMap.h"

using namespace bridges;

// This tutorial illustrates how to access the data of the US map with state
// boundaries with different colors for states and its boundaries
// It also illustrates how to draw all the counties 
int main(int argc, char **argv) {

	// create bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	// set title
	bridges.setTitle("Tutorial: Drawing US Map with State Boundaries, and State Counties");
	bridges.setDescription("All US states, All Counties");

	// create a data source object
	DataSource ds(&bridges);
	
	// get us map data - all states and counties
	vector<bridges::dataset::USState> map_data = ds.getUSMapCountyData (); 

	// create a USMap object with the map data
	USMap us_maps(map_data);

	// set the data structure to the map
	bridges.setDataStructure(&us_maps);

	//visualize
	bridges.visualize();
	
	return 0;
}
